src/backend/distributed/planner/recursive_planning.c

```diff
@@ -2392,6 +2445,9 @@ BuildReadIntermediateResultsQuery(List *targetEntryList, List *columnAliasList,
 	Query *resultQuery = makeNode(Query);
 	resultQuery->commandType = CMD_SELECT;
 	resultQuery->rtable = list_make1(rangeTableEntry);
+#if PG_VERSION_NUM >= PG_VERSION_16
```

Comment:
should we set perminfoindex = 0 at some point in this function?

Reply:
No need, because rangeTableEntry has been created using makeNode, which means its perminfoindex already has default value of 0.

Reply:
mostly for readability, but this is also fine